xml/System.Text.Json/JsonSerializerOptions.xml

```diff
@@ -118,7 +118,11 @@ Once serialization or deserialization occurs, the list cannot be modified.
 
 ## Remarks
 
-The default buffer size, in bytes, is 16384.
+The default buffer size, in bytes, is 16,384.
+For most workloads, the default size is a reasonable amount of JSON to buffer while reading from a stream or writing to a stream.
+That is, it performs well without creating objects on the Large Object Heap for the Garbage Collector (GC) to track.
+We recommend that you leave this value unchanged unless changing it makes an observable difference in performance.
+For example, if your JSON is always in small payloads, you could run tests with the default buffer size at 4,096 to find out if performance improves significantly.
```

Comment:
We can consider adding something like this as well.

Conversely, if your JSON is known to be really large and in memory (such as returned as string or UTF-8 bytes rather than writing to a stream), setting the default buffer size to a large upper bound could help improve performance.

Reply:
I don't think decreasing it will improve performance in any normal scenario unless the available memory is extremely low (since 16K is not very much). However @ahsonkhan suggestion of increasing it in non-streaming scenarios is a known benefit for JSON sizes that are known to be large.